interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/component/defaultconfig/DefaultComponentConfigDeterminer.scala

```diff
@@ -48,22 +48,31 @@ object DefaultComponentConfigDeterminer {
     }
     SingleComponentConfig(
       params = None,
-      Some(DefaultsComponentIcon.forBuiltInComponent(info)),
+      icon = Some(DefaultsComponentIcon.forBuiltInComponent(info)),
       // TODO: move from defaultModelConfig.conf to here + convention instead of code
       docsUrl = None,
       componentGroup = Some(componentGroup),
       componentId = Some(ComponentId.forBuiltInComponent(info))
     )
   }
 
-  def forFragment(componentId: Option[ComponentId], docsUrl: Option[String]): SingleComponentConfig =
-    SingleComponentConfig(
-      params = None,
-      Some(DefaultsComponentIcon.FragmentIcon),
+  // For fragments, we don't need to return SingleComponentConfig, because this config won't merged with anything else
+  // We can just return final, ComponentUiDefinition
+  def forFragment(
+      docsUrl: Option[String],
+      translateGroupName: ComponentGroupName => Option[ComponentGroupName],
```

Comment:
The other way would be to validate group name mapping against the fact that nobody translates "fragments" group to this hidden group but TBH I don't want to do a lot of engineering around this because I'm not sure if we  still want to keep the support for this hidden null group. I think that it is a legacy from times when built-in components could't be disabled. Now they can be disabled.